internal/querywatcher/query_watcher.go

```diff
@@ -37,11 +40,12 @@ type (
 		ResponseChan chan AdhocQueryResult
 	}
 
-	// QueryWatcher watches for changes in keys and notifies clients.
-	QueryWatcher struct {
-		WatchList    sync.Map                       // WatchList is a map of queries to their respective clients, type: map[DSQLQuery]*sync.Map[int]struct{}
-		QueryCache   *swiss.Map[string, cacheStore] // QueryCache is a map of queries to their respective data caches
-		QueryCacheMu sync.RWMutex
+	// QWatchManager watches for changes in keys and notifies clients.
+	QWatchManager struct {
+		WatchList        sync.Map                               // WatchList is a map of fingerprints to their respective clients, type: map[string]*sync.Map[int]struct{}
+		QueryCache       *swiss.Map[string, cacheStore]         // QueryCache is a map of fingerprints to their respective data caches
+		FingerPrintCache *swiss.Map[string, map[int]*DSQLQuery] // FingerPrintCache is a map of fingerprint to the respective query with the ClientFD
```

Comment:
The fingerprinting mechanism is probably internal to the query watcher, we can make it private.
Also maybe a better name for this is `fingerprintToQueryMap`

Reply:
I removed this with the change to keep Watchlist at the query level; hence, we don't need the Fingerprint cache.